python/ray/serve/_private/application_state.py

```diff
@@ -423,18 +424,28 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
         if self._target_state.deleting:
             return ApplicationStatus.DELETING, ""
 
-        num_healthy_deployments = 0
+        num_healthy_or_autoscaling_deployments = 0
+        num_updating_deployments = 0
         unhealthy_deployment_names = []
 
         for deployment_status in self.get_deployments_statuses():
             if deployment_status.status == DeploymentStatus.UNHEALTHY:
                 unhealthy_deployment_names.append(deployment_status.name)
-            if deployment_status.status == DeploymentStatus.HEALTHY:
-                num_healthy_deployments += 1
+            elif (
+                deployment_status.status == DeploymentStatus.HEALTHY
+                or deployment_status.status_trigger
+                == DeploymentStatusTrigger.AUTOSCALING
+            ):
+                # If all deployments are HEALTHY or autoscaling, then
+                # application is RUNNING
+                num_healthy_or_autoscaling_deployments += 1
+            else:
+                # If deployments are UPDATING or UPSCALING/DOWNSCALING
+                # with status trigger CONFIG_UPDATE, then application is
+                # still DEPLOYING
+                num_updating_deployments += 1
```

Comment:
What if deployment is UPDATING and status_trigger == AUTOSCALING, how are we planning to count such Deployment?

Reply:
That shouldn't be possible. If status trigger is autoscaling, then the updated target state should only have num replicas changed, so status will be set to upscaling/downscaling.

Reply:
Got it. Why do we need to mix it into this conditional then?

Reply:
I've separated it out into a separate case